sbatch_scripts/gen_sbatch.py

```diff
@@ -0,0 +1,406 @@
+        params["training-data-folder"] = args.training_data_folder
+        params["network-id"] = args.network_id
+        params["dl-workers"] = args.dl_workers
+
+    params["log-level"] = args.log_level
+
+    return params
+
+def main():
+
+    # Setting up argument parsing 
+
+    description = __doc__ # docstring for gen_sbatch
+    prog = "gen_sbatch" # program name
+
+    epilog = (
+        f"Example:\n    {prog} generate --config-path path/to/config.yaml --output-path path/to/output/folder --time 24:00:00 --sh-only --make-env --log-level INFO\n"
+        f"    {prog} jaxtrain --config-path path/to/config.yaml --output-path path/to/trained_network_output --training-data-folder path/to/training_data --network-id 0 --dl-workers 4 --time 00:10:00 --sh-only --make-env --log-level INFO\n" # examples for generate and jaxtrain scripts
+    )
+    parser = argparse.ArgumentParser(
+        description = description,
+        prog = prog,
+        epilog = epilog,
+        formatter_class=RawDescriptionHelpFormatter 
+    )
+
+    # Only the config-path and --output-path arguments, since these should be the first two arguments for generate and jaxtrain
+    parent_parser_config = argparse.ArgumentParser(add_help=False)
+    parent_parser_config.add_argument(
+        "--config-path",
+        default="dump",
+        help="Path to configuration .yaml file for running commands (default: 'dump')",
+        type=Path,
+        required=True
+    )
+    parent_parser_config.add_argument(
+        "--output-path",
+        default="dump",
+        help="Path to output folder for simulated data (generate) or trained neural network (jaxtrain)",
+        type=Path,
+        required=True
+    )
+    
+    # Common metadata arguments for both generate and jaxtrain
+    parent_parser_metadata = argparse.ArgumentParser(add_help=False)
+    parent_parser_metadata.add_argument(
+        "--time", 
+        help="Wall time limit for each job (default: 24:00:00)", 
+        default="24:00:00"
+    )
+    parent_parser_metadata.add_argument(
+        "--sh-only",
+        action="store_true",
+        help="Generate the sbatch script without submitting the job",
+    )
+    parent_parser_metadata.add_argument(
+        "--make-env",
+        action="store_true",
+        help="Create correct environment to run generate or jaxtrain in the SBATCH script",
+    )
+    parent_parser_metadata.add_argument(
+        "--log-level",
+        default="WARNING",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        help="Set the log level (default=WARNING)",
+    )
+
+    # Subparsers for arguments specific to generate and jaxtrain
+    subparsers = parser.add_subparsers(
+        title="commands",
+        dest="command",
+        required=True,
+    )
+    # Generate args
+    generate_parser = subparsers.add_parser(
+        "generate", help="Generates simulated data from model parameters", 
+        parents=[parent_parser_config]
+    )
+
```

Comment:
Add the arg to the generate parser.
```suggestion

    generate_parser.add_argument(
        "--array-size", type=int, default=1, help="Size of the job array (default: 1)"
    )
```